thrimshim/thrimshim/main.py

```diff
@@ -46,13 +46,50 @@ def _start_response(status, headers, exc_info=None):
 		return app(environ, _start_response)
 	return handle
 
+def authenticate_artist(f):
+	""""Authenticate a token against the database to authenticate an artist
 
-def authenticate(f):
-	"""Authenticate a token against the database.
+	Reference: https://developers.google.com/identity/sign-in/web/backend-auth"""
+    
+	@wraps(f)
+	def artist_auth_wrapper(*args, **kwargs):
```

Comment:
Rather than repeating this entire function with `artist` instead of `editor`, could we have one version of the function and pass in the role that we want to test?

Reply:
This is me not understanding decorators very well and being too lazy to figure out how to write a decorator that you can pass an argument to.

Reply:
You could still have two seperate decorators that just call out to a common function.
Or if you want a decorator that takes an argument, here is the standard idiom:
```
def mydecorator(arg):
    def mydecorator_generated(f):
        @wraps(f)
        def mydecorator_wrapper(*args, **kwargs):
            ...
        return mydecorator_wrapper
    return mydecorator_generated
```
It's a bit convoluted but the idea is that when you call `mydecorator(arg)`, it returns the actual decorator which has the arg "baked in", and the decorator then returns the wrapped function.

You'd then use it like:
```
@mydecorator(arg)
def wrapped_func():
    ...
```

Reply:
I am going with two separate decorators that share a single function.